numpy/core/tests/test_multiarray.py

```diff
@@ -2326,6 +2326,22 @@ def test_dot_type_mismatch(self):
         assert_raises(TypeError, np.dot, c, A)
         assert_raises(TypeError, np.dot, A, c)
 
+    def test_dot_out_mem_overlap(self):
+        np.random.seed(1)
+
+        for dtype in [np.object_, np.float32, np.complex128, np.int64]:
```

Comment:
do we need multiple dtype tests at all? none of the new code has any relations to type. Doesn't matter much here as its a fast test, but I like to avoid unnecessary test bloat.